docs/backend-system/core-services/url-reader.md

```diff
@@ -45,3 +45,99 @@ createBackendPlugin({
   },
 });
 ```
+
+## Providing custom URL readers
+
+Custom URL readers that are not intended for open sourcing via a custom service factory for that specific reader. The following example shows how to create a custom URL reader and provide it to the backend.
```

Comment:
First sentence seems like it's missing something. "If you want to ..." or something to start it off?